src/test/java/org/mastodon/mamut/feature/branch/BranchNSubBranchSpotsFeatureComputerTest.java

```diff
@@ -0,0 +1,201 @@
+	 *                                            Spot( 10, X=0,00, Y=0,00, Z=0,00, tp=0 )  Spot( 7, X=0,00, Y=0,00, Z=0,00, tp=0 )
+	 * </pre>
+	 * <h1>BranchSpot-Graph</h1>
+	 * <pre>
+	 *                        branchSpot0
+	 * 	       ┌──────────────┴─────────────────┐
+	 * 	       │                                │
+	 * 	   branchspot1                      branchSpot2
+	 * 	┌──────┴───────┐
+	 * 	│              │
+	 * branchspot3 branchSpot4
+	 * </pre>
+	 */
+	@Test
+	public void testComputeNumberOfSubtreeNodes1()
+	{
+		try(Context context = new Context())
+		{
+			final MamutFeatureComputerService featureComputerService = context.getService( MamutFeatureComputerService.class );
+			Model model = new Model();
+			ModelGraph modelGraph = model.getGraph();
+
+			Spot spot0 = addNode( modelGraph, "0" );
+			Spot spot1 = addNode( modelGraph, "1" );
+			Spot spot2 = addNode( modelGraph, "2" );
+			Spot spot3 = addNode( modelGraph, "3" );
+			Spot spot4 = addNode( modelGraph, "4" );
+			Spot spot5 = addNode( modelGraph, "5" );
+			Spot spot6 = addNode( modelGraph, "6" );
+			Spot spot7 = addNode( modelGraph, "7" );
+			Spot spot8 = addNode( modelGraph, "8" );
+			Spot spot9 = addNode( modelGraph, "9" );
+			Spot spot10 = addNode( modelGraph, "10" );
+			Spot spot11= addNode( modelGraph, "11" );
+			Spot spot12 = addNode( modelGraph, "12" );
+			Spot spot13 = addNode( modelGraph, "13" );
+
+			modelGraph.addEdge( spot0, spot1 );
+			modelGraph.addEdge( spot1, spot2 );
+			modelGraph.addEdge( spot2, spot3 );
+			modelGraph.addEdge( spot3, spot4 );
+			modelGraph.addEdge( spot4, spot5 );
+			modelGraph.addEdge( spot5, spot6 );
+			modelGraph.addEdge( spot6, spot7 );
+			modelGraph.addEdge( spot4, spot8 );
+			modelGraph.addEdge( spot8, spot9 );
+			modelGraph.addEdge( spot9, spot10 );
+			modelGraph.addEdge( spot2, spot11 );
+			modelGraph.addEdge( spot11, spot12 );
+			modelGraph.addEdge( spot12, spot13 );
+
+			model.getBranchGraph().graphRebuilt();
+			featureComputerService.setModel( model );
+			Feature<BranchSpot> feature = (Feature< BranchSpot> ) featureComputerService.compute( BranchNSubBranchSpotsFeature.SPEC ).get(BranchNSubBranchSpotsFeature.SPEC);
+
+			ModelBranchGraph modelBranchGraph = model.getBranchGraph();
+			BranchSpot ref = modelBranchGraph.vertexRef();
+			assertEquals( 5, modelBranchGraph.vertices().size() );
+
+			BranchSpot branchSpot0 = modelBranchGraph.getBranchVertex( spot0, modelBranchGraph.vertexRef() );
+			assertEquals( 2, branchSpot0.outgoingEdges().size());
+			assertEquals(4, feature.project( FeatureProjectionKey.key( new FeatureProjectionSpec( BranchNSubBranchSpotsFeature.KEY ) ) ).value( branchSpot0 ), 0);
+
+			BranchSpot branchSpot1 = modelBranchGraph.getBranchVertex( spot4, modelBranchGraph.vertexRef() );
+			assertEquals( 1, branchSpot1.incomingEdges().size());
+			assertEquals( 2, branchSpot1.outgoingEdges().size());
```

Comment:
It's not the purpose of this test to check the number of incoming and outgoing edges.
The test would benefit from not having these asserts. It would be more streamlined and less confusing.

Reply:
Okay. Removed the tests for the edges.